### PR TITLE
Tests / TS infrastructure / Util methods

### PR DESCRIPTION
**Pull request by lukiano**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,13 +1,15 @@
 {
   "name": "aff",
-  "version": "0.2.1",
+  "version": "0.2.0",
```
Comment (srijs) [inline package.json]: Looks like you might need to rebase your changes.

File: src/core/eff-spec.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+import * as mocha from 'mocha';
+import * as chai from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+
+import {Op, Eff} from './eff';
+import {Run} from './run';
+import {EffUtil} from './effutil';
+
+chai.should();
```
Comment (srijs) [inline src/core/eff-spec.ts]: Would it be possible to use the `expect` interface of `chai`, which doesn't monkey-patch `Object.prototype`?

File: src/core/effutil.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    promise: Promise<T> = new Promise<T>((resolve, reject) => {
+        this.resolve = resolve;
+        this.reject = reject;
+    });
+    resolve: (t: T) => void;
+    reject: (e: Error) => void;
+}
+
+export class EffUtil {
+    static scheduledOnce<T>(block: () => T, delay: number): Eff<{}, T> {
```
Comment (srijs) [inline src/core/effutil.ts]: I think this (and the other utilities) should be "for all effects", so this should look more like `scheduledOnce<F, T>(block: () => T, delay: number): Eff<F, T>`.

File: src/core/run.ts [hunk 4]
```diff
@@ -8,7 +8,7 @@ export class Run<T> {
   }
 
   static fail<T>(err: Error): Run<T> {
-    return new Run(Promise.reject(err) as Promise<any>, () => null);
+    return new Run(Promise.reject<T>(err), () => null);
```
Comment (srijs) [inline src/core/run.ts]: Nice one! Didn't know that trick yet.

File: package.json [hunk 5]
```diff
@@ -1,13 +1,15 @@
...
   "description": "Asynchronous effect system for TypeScript",
   "main": "out/index.js",
   "scripts": {
     "build": "tsc",
+    "install": "typings install",
+    "prepublish": "npm run build",
     "pretest": "npm run build",
-    "test": "mocha out/**/*-spec.js",
-    "prepublish": "npm run build"
+    "test": "./node_modules/.bin/mocha out/**/*-spec.js --recursive",
```
Comment (srijs) [inline package.json]: Did just `mocha` instead of `./node_modules/.bin/mocha` not work for you?

File: package.json [hunk 6]
```diff
@@ -1,13 +1,15 @@
 {
   "name": "aff",
-  "version": "0.2.1",
+  "version": "0.2.0",
   "description": "Asynchronous effect system for TypeScript",
   "main": "out/index.js",
   "scripts": {
     "build": "tsc",
+    "install": "typings install",
```
Comment (srijs) [inline package.json]: I think we need to add `typings` to the `devDependencies` as well.

File: package.json (hunk 5 above)
Comment (lukiano) [inline package.json]: mocha works, but it's an older version. With that command I make sure it's the version set in package.json

File: package.json (hunk 1 above)
Comment (lukiano) [inline package.json]: I can't seem to rebase so I'll add a new commit with the version update

File: package.json (hunk 5 above)
Comment (srijs) [inline package.json]: Hm, I was under the impression that `npm` prepended `./node_modules/.bin/` to the `PATH` before executing scripts.

Comment (srijs) [inline package.json]: https://docs.npmjs.com/misc/scripts#path

Comment (lukiano) [inline package.json]: awesome

File: package.json (hunk 6 above)
Comment (lukiano) [inline package.json]: done

File: src/core/effutil.ts (hunk 3 above)
Comment (lukiano) [inline src/core/effutil.ts]: done

**Pull request merged**